teleportr/drivers/disburser/driver.go

```diff
@@ -140,8 +150,8 @@ func (d *Driver) ClearPendingTx(
 func (d *Driver) GetBatchBlockRange(
 	ctx context.Context) (*big.Int, *big.Int, error) {
 
-	// Update balance metrics on each iteration.
-	d.updateBalanceMetrics(ctx)
+	// Update metrics on each iteration.
+	d.collectChainMetrics(ctx)
```

Comment:
Now that there's a bg gorotuine also doing this, would it be better to only send a signal to `collectChainMetricsBackground` for metrics to be immediately collected?

Reply:
I feel like this is a bit cleaner. The goroutine just wraps this method. Would like to avoid having to do signal lifecycle management.